firefox_puppeteer/ui/toolbars.py

```diff
@@ -38,13 +42,18 @@ def home_button(self):
         """
         return self.marionette.find_element(By.ID, 'home-button')
 
-    @use_class_as_property('ui.toolbars.LocationBar')
+    @property
     def locationbar(self):
         """Provides access to the DOM elements contained in the
         locationbar.
 
         See the :class:`LocationBar` reference.
         """
+        if not self._locationbar:
+            from .toolbars import LocationBar
+            self._locationbar = LocationBar(lambda: self.marionette, self.window, self.element)
```

Comment:
The last parameter for the element has to be the DOM node for the location bar itself. Currently you can find that inside the LocationBar class as [`urlbar`](https://github.com/mozilla/firefox-ui-tests/blob/master/firefox_puppeteer/ui/toolbars.py#L215) property. Call the `find_element()` method before this line and pass in the element. Similar to what I do here: https://github.com/mozilla/firefox-ui-tests/blob/master/firefox_puppeteer/ui/windows.py#L473